src/Dapr.Client/DaprClientGrpc.cs

```diff
@@ -1373,6 +1375,563 @@ private Autogenerated.StateItem ToAutogeneratedStateItem(StateTransactionRequest
+                    yield return Array.Empty<byte>();
+                }
+            }
+
+            //Send the completion message
+            await duplexStream.RequestStream.CompleteAsync();
+            yield break; //End the async enumeration
+        }
+
+        /// <summary>
+        /// Retrieves the encrypted bytes from the encryption operation on the sidecar and returns as an enumerable stream.
+        /// </summary>
+        private async IAsyncEnumerable<byte[]> RetrieveEncryptedStreamAsync(AsyncDuplexStreamingCall<Autogenerated.EncryptRequest, Autogenerated.EncryptResponse> duplexStream, [EnumeratorCancellation] CancellationToken cancellationToken)
+        {
+            await foreach (var encryptResponse in duplexStream.ResponseStream.ReadAllAsync(cancellationToken)
+                               .ConfigureAwait(false))
+            {
+                yield return encryptResponse.Payload.Data.ToByteArray();
+            }
+        }
+        
+        /// <summary>
+        /// Decrypts the specified stream of ciphertext using the Dapr Cryptography encryption functionality.
+        /// </summary>
+        /// <param name="vaultResourceName">The name of the vault resource used by the operation.</param>
+        /// <param name="ciphertextStream">The stream containing the bytes of the ciphertext value to decrypt.</param>
+        /// <param name="keyName">The name of the key to use from the Vault for the decryption operation.</param>
+        /// <param name="decryptionOptions">Options informing how the decryption operation should be configured.</param>
+        /// <param name="cancellationToken">A <see cref="CancellationToken"/> that can be used to cancel the operation.</param>
+        /// <returns>An asynchronously enumerable array of decrypted bytes.</returns>
+        public override async IAsyncEnumerable<byte[]> DecryptStreamAsync(string vaultResourceName, Stream ciphertextStream, string keyName,
+            DecryptionOptions decryptionOptions, [EnumeratorCancellation] CancellationToken cancellationToken = default)
+        {
+            ArgumentVerifier.ThrowIfNullOrEmpty(vaultResourceName, nameof(vaultResourceName));
+            ArgumentVerifier.ThrowIfNullOrEmpty(keyName, nameof(keyName));
+            ArgumentVerifier.ThrowIfNull(ciphertextStream, nameof(ciphertextStream));
+            ArgumentVerifier.ThrowIfNull(decryptionOptions, nameof(decryptionOptions));
+
+            var decryptRequestOptions = new Autogenerated.DecryptRequestOptions
+            {
+                ComponentName = vaultResourceName, KeyName = keyName
+            };
+
+            var options = CreateCallOptions(headers: null, cancellationToken);
+            var duplexStream = client.DecryptAlpha1(options);
+
+            //Run both operations at the same time, but return the output of the streaming values coming from the operation
+            var tasks = new Task<IAsyncEnumerable<byte[]>>[]
```

Comment:
I feel like this might be more complicated than it need to (in particular, requiring `SendCiphertextStreamAsync()` return an async enumerable just so that it can be grouped together with the parallel task, and then having to iterate and re-yield the retrieved async enumerable).  Might something like this work:

```csharp
var sendTask = SendCiphertextStreamAsync(ciphertextStream, (int)decryptionOptions.StreamingBlockSizeInBytes, duplexStream, decryptRequestOptions, cancellationToken);

var retrieveTask = RetrieveDecryptedStreamAsync(duplexStream, cancellationToken);

await Task.WhenAll(sendTask, retrieveTask)

return await retrieveTask;
```

While it's technically not necessary to `await` the send task, as the retrieve task shouldn't effectively complete unless the send tasks completes, it still valid to do so (and may avoid Roslyn warnings about tasks not being awaited).

Reply:
Whoops, don't code tired - forgot to even apply the `Task.WhenAll` in there so this never would have even sent anything. Updated and simplified according to your suggestion.

Reply:
Whoops - introduced a regression here since I wasn't even awaiting the tasks I put in the array. Fixed and simplified per your suggestion.